pkg/controllers/daemon/service/controller.go

```diff
@@ -74,9 +74,12 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 		return ctrl.Result{}, nil
 	}
+	// ignore headless services
+	if service.Spec.ClusterIP == "None" {
+		return ctrl.Result{}, nil
+	}
 	ips := retinaCommon.IPAddresses{}
 	ips.IPv4 = net.ParseIP(service.Spec.ClusterIP)
-	net.ParseIP(service.Spec.ClusterIP)
```

Comment:
any idea why this line was here?

Reply:
No but I removed it as it is not necessary - according to the doc `net.ParseIP` returns a `net.IP` which was not assigned to anything in the line I removed.

```go
func net.ParseIP(s string) net.IP
ParseIP parses s as an IP address, returning the result. The string s can be in IPv4 dotted decimal ("192.0.2.1"), IPv6 ("2001:db8::68"), or IPv4-mapped IPv6 ("::ffff:192.0.2.1") form. If s is not a valid textual representation of an IP address, ParseIP returns nil.
```